File: Android/app/src/main/java/org/gdgsp/model/Event.java
/*
 * Copyright (C) 2016 Alefe Souza <http://alefesouza.com>
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package org.gdgsp.model;

import java.io.Serializable;
import java.util.List;

/**
 * Objeto que representa um evento no aplicativo.
 */
public class Event implements Serializable {
	private int id, yes_rsvp_count, rsvp_limit, waitlist_count;
	private double lat, lon;
	private String name, image, description, link, who, place, address, city, start, end, response;
	private boolean rsvpable;
	private List<Question> survey_questions, answers;

	public int getId() {
		return id;
	}

	public void setId(int id) {
		this.id = id;
	}

	public int getYes_rsvp_count() {
		return yes_rsvp_count;
	}

	public void setYes_rsvp_count(int yes_rsvp_count) {
		this.yes_rsvp_count = yes_rsvp_count;
	}

	public int getRsvp_limit() {
		return rsvp_limit;
	}

	public void setRsvp_limit(int rsvp_limit) {
		this.rsvp_limit = rsvp_limit;
	}

	public int getWaitlist_count() {
		return waitlist_count;
	}

	public void setWaitlist_count(int waitlist_count) {
		this.waitlist_count = waitlist_count;
	}

	public double getLat() {
		return lat;
	}

	public void setLat(double lat) {
		this.lat = lat;
	}

	public double getLon() {
		return lon;
	}

	public void setLon(double lon) {
		this.lon = lon;
	}

	public String getName() {
		return name;
	}

	public void setName(String name) {
		this.name = name;
	}

	public String getImage() {
		return image;
	}

	public void setImage(String image) {
		this.image = image;
	}

	public String getDescription() {
		return description;
	}

	public void setDescription(String description) {
		this.description = description;
	}

	public String getLink() {
		return link;
	}

	public void setLink(String link) {
		this.link = link;
	}

	public String getWho() {
		return who;
	}

	public void setWho(String who) {
		this.who = who;
	}

	public String getPlace() {
		return place;
	}

	public void setPlace(String place) {
		this.place = place;
	}

	public String getAddress() {
		return address;
	}

	public void setAddress(String address) {
		this.address = address;
	}

	public String getCity() {
		return city;
	}

	public void setCity(String city) {
		this.city = city;
	}

	public String getStart() {
		return start;
	}

	public void setStart(String start) {
		this.start = start;
	}

	public String getEnd() {
		return end;
	}

	public void setEnd(String end) {
		this.end = end;
	}

	public String getResponse() {
		return response;
	}

	public void setResponse(String response) {
		this.response = response;
	}

	public boolean isRsvpable() {
		return rsvpable;
	}

	public void setRsvpable(boolean rsvpable) {
		this.rsvpable = rsvpable;
	}

	public List<Question> getSurvey_questions() {
		return survey_questions;
	}

	public void setSurvey_questions(List<Question> survey_questions) {
		this.survey_questions = survey_questions;
	}

	public List<Question> getAnswers() {
		return answers;
	}

	public void setAnswers(List<Question> answers) {
		this.answers = answers;
	}
}

File: Back-end/GDGSP/Model/Event.php
<?php
/*
 * Copyright (C) 2017 Alefe Souza <contact@alefesouza.com>
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

namespace GDGSP\Model {
    class Event extends JsonCommon {
        protected $id, $name, $image, $description, $link, $who, $place, $address, $city, $lat, $lon, $start, $end, $yes_rsvp_count, $rsvp_limit, $waitlist_count, $response, $survey_questions, $answers, $rsvpable, $how_to_find_us, $image_width, $image_height;

        public function __construct(int $id, string $name, string $image, string $description, string $link, string $who, string $place, string $address, string $city, float $lat, float $lon, string $start, string $end, int $yes_rsvp_count, int $rsvp_limit, int $waitlist_count, string $response, array $survey_questions, $answers, bool $rsvpable, string $how_to_find_us) {
            $this->id = $id;
            $this->name = $name;
            $this->image = $image;
            $this->description = $description;
            $this->link = $link;
            $this->who = $who;
            $this->place = $place;
            $this->address = $address;
            $this->city = $city;
            $this->lat = $lat;
            $this->lon = $lon;
            $this->start = $start;
            $this->end = $end;
            $this->yes_rsvp_count = $yes_rsvp_count;
            $this->rsvp_limit = $rsvp_limit;
            $this->waitlist_count = $waitlist_count;
            $this->response = $response;
            $this->survey_questions = $survey_questions;
            $this->answers = $answers;
            $this->rsvpable = $rsvpable;
            $this->how_to_find_us = $how_to_find_us;
        }

        public function setId(int $id) { $this->id = $id; }
        public function getId() : int { return $this->id; }
        public function setName(string $name) { $this->name = $name; }
        public function getName() : string { return $this->name; }
        public function setImage(string $image) { $this->image = $image; }
        public function getImage() : string { return $this->image; }
        public function setDescription(string $description) { $this->description = $description; }
        public function getDescription() : string { return $this->description; }
        public function setLink(string $link) { $this->link = $link; }
        public function getLink() : string { return $this->link; }
        public function setWho(string $who) { $this->who = $who; }
        public function getWho() : string { return $this->who; }
        public function setPlace(string $place) { $this->place = $place; }
        public function getPlace() : string { return $this->place; }
        public function setAddress(string $address) { $this->address = $address; }
        public function getAddress() : string { return $this->address; }
        public function setCity(string $city) { $this->city = $city; }
        public function getCity() : string { return $this->city; }
        public function setLat(float $lat) { $this->lat = $lat; }
        public function getLat() : float { return $this->lat; }
        public function setLon(float $lon) { $this->lon = $lon; }
        public function getLon() : float { return $this->lon; }
        public function setStart(string $start) { $this->start = $start; }
        public function getStart() : string { return $this->start; }
        public function setEnd(string $end) { $this->end = $end; }
        public function getEnd() : string { return $this->end; }
        public function setYes_rsvp_count(int $yes_rsvp_count) { $this->yes_rsvp_count = $yes_rsvp_count; }
        public function getYes_rsvp_count() : int { return $this->yes_rsvp_count; }
        public function setRsvp_limit(int $rsvp_limit) { $this->rsvp_limit = $rsvp_limit; }
        public function getRsvp_limit() : int { return $this->rsvp_limit; }
        public function setWaitlist_count(int $waitlist_count) { $this->waitlist_count = $waitlist_count; }
        public function getWaitlist_count() : int { return $this->waitlist_count; }
        public function setResponse(string $response) { $this->response = $response; }
        public function getResponse() : string { return $this->response; }
        public function setSurvey_questions(array $survey_questions) { $this->survey_questions = $survey_questions; }
        public function getSurvey_questions() : array { return $this->survey_questions; }
        public function setAnswers($answers) { $this->answers = $answers; }
        public function getAnswers() { return $this->answers; }
        public function setRsvpable(bool $rsvpable) { $this->rsvpable = $rsvpable; }
        public function getRsvpable() : bool { return $this->rsvpable; }
        public function setHow_to_find_us(string $how_to_find_us) { $this->how_to_find_us = $how_to_find_us; }
        public function getHow_to_find_us() : string { return $this->how_to_find_us; }
        public function setImage_width(int $image_width) { $this->image_width = $image_width; }
        public function getImage_width() : int { return $this->image_width; }
        public function setImage_height(int $image_height) { $this->image_height = $image_height; }
        public function getImage_height() : int { return $this->image_height; }
    }
}
?>